firedrake/variational_solver.py

```diff
@@ -298,7 +298,7 @@ def __init__(self, a, L, u, bcs=None, aP=None,
         # In the linear case, the Jacobian is the equation LHS.
         J = a
         # Jacobian is checked in superclass, but let's check L here.
-        if L is 0:  # noqa: F632
+        if L == 0:  # noqa: F632
```

Comment:
Nope, because `__eq__` is overloaded on forms.

`if not isinstance(L, ufl.Form) and L == 0`